### PR TITLE
Rebuild treadstone48 rev1 firmware

### PR DESCRIPTION
**Pull request by marksard**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

I re-created treastone48 keyboard firmware from ./util/new_keyboard.sh
After that I use common OLED_DRIVER, RGB_LIGHT and SPLIT_KEYBOARD options.
And then check all keymap.
This PR is point at removed Helix/crkbd based source code.

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/treadstone48/keymaps/default/keymap.c [hunk 1]
```diff
@@ -1,7 +1,23 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
 #include QMK_KEYBOARD_H
 #include "keymap_jp.h"
 #include "../common/oled_helper.h"
 
+extern keymap_config_t keymap_config;
```
Comment (fauxpark) [inline keyboards/treadstone48/keymaps/default/keymap.c]:
```suggestion
```

File: keyboards/treadstone48/rev1/rev1.c [hunk 2]
```diff
@@ -1,15 +1,50 @@
...
+}
 
-	matrix_init_user();
-};
+bool led_update_kb(led_t led_state) {
+    // put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
 
+    return led_update_user(led_state);
+}
+*/
```
Comment (fauxpark) [inline keyboards/treadstone48/rev1/rev1.c]: All of this can be deleted.

File: keyboards/treadstone48/rules.mk [hunk 3]
```diff
@@ -11,31 +11,39 @@ MCU = atmega32u4
...
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+# Otherwise, delete this section
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (fauxpark) [inline keyboards/treadstone48/rules.mk]: Please remove this block.

**Comment by marksard** [comment]
thanks

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**